### ISSUE TITLE
How to detect whether the gnuplot terminal is still running?

### ISSUE THREAD

**Issue by wucke13** [SiegeLord/RustGnuplot#62]
I want my rust code to `loop {}` until the plot window is closed. How do I do this?

This is needed, as most gnuplot terminals loose control facilities if the their gnuplot process dies, which always happens if the rust program which spawned gnuplot dies. So in order to have that rust program wait for the gnuplot terminal to terminate, one needs to know whether gnuplot is still running.

**Comment by SiegeLord** [comment]
Currently we interact with gnuplot as follows:

1. We spawn it using its `--persist` flag.
2. We issue the `quit` command when `Figure` is dropped.

(2) was added to kill zombie `gnuplot` processes that (1) sometimes created. It sounds like we need this to be configurable, either by altering (1) or (2).

I just experimented with this a little bit, and this appears to work:

1. Issue `pause mouse close`
2. Issue `quit`.
3. Check if `gnuplot` is closed via `Process::try_wait`.

If you don't do (1), then `gnuplot` exits right away. If you don't do (2), `gnuplot` never exits.

I guess (1) can be done via `Figure::set_post_commands`, so the only thing missing is a version of `Figure::close` that does a `try_wait` rather than `wait`.

**Comment by wucke13** [comment]
Hi & thank you for the detailed analysis. I'm fine with whatever solution you prefer. However, a more intuitive solution than having to set the pre and post commands would be preferabl IMHO. I think the best would be a function which just tells whether gnuplot is still running. This would allow for a non blocking wait on gnuplot.

**Comment by SiegeLord** [comment]
Yeah, I'll see what can be done. Another issue is that even if I fix this, some things still fail to work correctly with the UI (like multiplot).

**Comment by SiegeLord** [comment]
With 52016af, `show` now returns a `CloseSentinel` that you can call `wait` or `try_wait` to determine if the `gnuplot` process is still running.
